src/components/Results/index.js

```diff
@@ -12,10 +12,20 @@ var paginationLabels = {
 class Results extends React.Component {
   render() {
     var results = this.props.results;
-    var renderedResults = map( results.hits, function( r ) {
-      return <Hogan data={ r } key={ r.objectID } template={ this.props.hitTemplate } />;
-    }, this );
-    return <div className="search_list search_results_container row">{ renderedResults }</div>
+    return  ( results && results.hits && results.hits.length > 0 ) ?
+              this.renderWithResults( results.hits, this.props.hitTemplate ) :
+              this.renderNoResults( results, this.props.noResultsTemplate );
+  }
+  renderWithResults( hits, hitTemplate ) {
+    var renderedHits = map( hits, function( hit ) {
+      return <Hogan data={ hit } key={ hit.objectID } template={ hitTemplate } />;
+    } );
+    return <div className="search_list search_results_container row">{ renderedHits }</div>
+  }
+  renderNoResults( results, noResultsTemplate ){
+    return  <div className="search_list search_results_container row">
+              <Hogan data={ results } template={ this.props.noResultsTemplate } />
```

Comment:
You're using the this.props.noResultsTemplate instead of the noResultsTemplate in the args.

Also, both methods could surely be merge in one (the map will be empty if no `hits`, and the template is injected anyway).